test/testenv/verificationutils.go

```diff
@@ -441,6 +441,36 @@ func VerifySecretsUpdatedOnPod(deployment *Deployment, testenvInstance *TestEnv,
 	}
 }
 
+// VerifyNewSecretValueOnVersionedSecretObject Check whether the new versioned secret object is created with new value
+func VerifyNewSecretValueOnVersionedSecretObject(deployment *Deployment, testenvInstance *TestEnv, verificationSecrets []string, secretKey string, previousValue string) {
+	for _, secretObject := range verificationSecrets {
+		found := false
+		currentValue := GetSecretKey(deployment, testenvInstance.GetName(), secretKey, secretObject)
+		if currentValue == previousValue {
+			testenvInstance.Log.Info("New Key Not created ", "Secret Object Name", secretObject, "Secret Key", secretKey, "Previous Value of Key", previousValue, "Key Value found", currentValue)
+		} else {
+			testenvInstance.Log.Info("New key created ", "Secret Object Name", secretObject, "Secret Key", secretKey, "Previous Value of Key", previousValue, "Key Value found", currentValue)
+			found = true
+		}
+		gomega.Expect(found).Should(gomega.Equal(true))
+	}
+}
+
+// VerifyNewVersionedSecretValueUpdatedOnPod Check whether the new secret object value is mounted on all pods
```

Comment:
`VerifySecretValueUpdatedOnPod` vs `VerifyNewVersionedSecretValueUpdatedOnPod` 
 Also change comment to say `verifies the given secret value is not present on pod.`

Reply:
modified